pkg/vm/engine/tae/tables/indexwrapper/bfnode.go

```diff
@@ -51,16 +51,29 @@ func NewBfReader(
 
 func (r *BfReader) getBloomFilter() (index.StaticFilter, error) {
 	// return r.reader.LoadOneBF(context.Background(), r.bfKey.ID())
-	v, ok := r.indexCache.Get(r.reader.GetName())
+	bs, ok := r.indexCache.Get(r.reader.GetName())
 	if ok {
-		return v.([]objectio.StaticFilter)[r.bfKey.ID()], nil
+		var obj any
```

Comment:
too heavy to access bf. I think I need a new lru cache w/o unmarshalling

Reply:
Right now the code is like this

```go
func (r *BfReader) getBloomFilter() (index.StaticFilter, error) {
	bs, ok := r.indexCache.Get(*r.blockID)
	if ok {
		bf, err := index.DecodeBloomFilter(bs)
		if err != nil {
			return nil, err
		}
		return bf, nil
	}

	v, size, err := r.reader.LoadOneBF(context.Background(), r.bfKey.ID())
	if err != nil {
		return nil, err
	}

	bs, err = v.Marshal()
	if err != nil {
		return nil, err
	}

	r.indexCache.Set(*r.blockID, bs, int64(size))
	return v, nil
}
```
We unmarshal the `bloom filter` every time we read from the cache. 

I don't have any comments on the performance impact.